CMakeLists.txt

```diff
@@ -71,10 +72,29 @@ include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
 set(TVCPP torchvision/csrc)
-list(APPEND ALLOW_LISTED ${TVCPP} ${TVCPP}/io/image ${TVCPP}/io/image/cpu ${TVCPP}/models ${TVCPP}/ops
-  ${TVCPP}/ops/autograd ${TVCPP}/ops/cpu ${TVCPP}/io/image/cuda)
+list(APPEND ALLOW_LISTED
+  ${TVCPP}
+  ${TVCPP}/io/image
+  ${TVCPP}/io/image/cpu
+  ${TVCPP}/models
+  ${TVCPP}/ops
+  ${TVCPP}/ops/autograd
+  ${TVCPP}/ops/cpu
+)
+
+# Add CUDA source files if requested.
 if(WITH_CUDA)
-    list(APPEND ALLOW_LISTED ${TVCPP}/ops/cuda ${TVCPP}/ops/autocast)
+    list(APPEND ALLOW_LISTED
+      ${TVCPP}/ops/cuda
+      ${TVCPP}/ops/autocast
+    )
+endif()
+
+# Add NVJPEG source files if requested.
+if(TORCHVISION_USE_NVJPEG)
```

Comment:
It is currently taken care by: 

https://github.com/pytorch/vision/blob/b3b74481a113610b33d27ab703733dad6d79bc91/torchvision/csrc/io/image/cuda/decode_jpeg_cuda.cpp#L5

If building without CUDA, the error would be shown at runtime. In my opinion adding the check for CUDA here would make sense, since it would give error at compile time if someone tries to use it

Reply:
I agree it'd be nice to have the error at compile time instead. Should we just move the `if(TORCHVISION_USE_NVJPEG)` block within the `if(WITH_CUDA)` block?

Reply:
Latest commit should solve it

Reply:
I'm not sure why but it looks like the build failed with

```
/usr/local/lib64/libtorchvision.so: undefined reference to `vision::image::decode_jpeg_cuda(at::Tensor const&, long, c10::Device)'
collect2: error: ld returned 1 exit status
```

https://app.circleci.com/pipelines/github/pytorch/vision/18800/workflows/69a3ed7f-b0e8-43ad-8340-6d070fea81e2/jobs/1521154

Reply:
I will try to fix it later today or tomorrow